src/main/groovy/io/seqera/wave/service/persistence/postgres/data/BuildRow.groovy

```diff
@@ -0,0 +1,49 @@
+/*
+ *  Wave, containers provisioning service
+ *  Copyright (c) 2023-2024, Seqera Labs
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.seqera.wave.service.persistence.postgres.data
+
+import java.time.Instant
+
+import groovy.transform.CompileStatic
+import io.micronaut.core.annotation.Introspected
+import io.micronaut.data.annotation.Id
+import io.micronaut.data.annotation.MappedEntity
+import io.micronaut.data.annotation.TypeDef
+import io.micronaut.data.model.DataType
+import jakarta.persistence.Table
+/**
+ * Model a row entry in the {@code wave_build} table
+ *
+ * @author Paolo Di Tommaso <paolo.ditommaso@gmail.com>
+ */
+@MappedEntity
+@Introspected
+@CompileStatic
+@Table(name = "wave_build")
+class BuildRow {
+
+    @Id
+    String id
+
+    @TypeDef(type = DataType.JSON)
+    String data  // JSONB data will be retrieved as a string
```

Comment:
I am not very familiar with this type of data structure when it comes to Java , reading some items on micronaut is there a particular reason for usig a String type here ? 

Asking for my own education as I was following this guide https://micronaut-projects.github.io/micronaut-data/4.6.0/guide/index.html#sqlJsonType

Reply:
I was not considering that because the target data structure should be a [WaveBuildRecord](https://github.com/seqeralabs/wave/blob/ea903fd247c0b47d3080adf989186adbdba38f1f/src/main/groovy/io/seqera/wave/service/persistence/WaveBuildRecord.groovy#L42-L42) object, therefore the idea is to load as a string and then deserialise into a `WaveBuildRecord` here https://github.com/seqeralabs/wave/blob/ea903fd247c0b47d3080adf989186adbdba38f1f/src/main/groovy/io/seqera/wave/service/persistence/postgres/PostgresPersistentService.groovy#L87

However i've just realised that declaring as 

```
    @TypeDef(type = DataType.JSON)
    WaveBuildRecord data
```


It manages the ser-deserialization automatically, neat!